torsvc/net.go

```diff
@@ -0,0 +1,58 @@
+package torsvc
+
+import (
+	"fmt"
+	"net"
+)
+
+// MultiNet is an implementation of the Net interface that abstracts away the
+// "net" and "torsvc" functions from the user. This can be used to "switch"
+// Tor functionality on/off based on the MultiNet.Tor boolean. MultiNet allows
+// for callers to call net functions when necessary and torsvc functions when
+// necessary
+type MultiNet struct {
+	Tor      bool
+	TorDNS   string
+	TorSocks string
+}
+
+// A compile time check to ensure MultiNet implements the Net interface.
+var _ Net = (*MultiNet)(nil)
+
+// Dial uses either the "net" or "torsvc" dial function.
+func (m *MultiNet) Dial(network, address string) (net.Conn, error) {
+	if m.Tor {
```

Comment:
This sort of defeats the entire purpose of having the interface. The interface is supposed to be a generic net interface with *multiple implementing structs*. So you would have the `Net` interface, then something like `type StandardNet struct` that implements it (which is a thin wrapper around the `net` package methods) and `type TorProxyNet struct` that implements it. https://en.wikipedia.org/wiki/Polymorphism_(computer_science)

Reply:
I'm thinking that I should remove the `Net` interface because otherwise I have to branch logic in a few places depending on whether `net.Dial` or `torsvc.TorDial` is used.  I forgot to delete the `Net` interface here and `MultiNet` def shouldn't implement it so thanks for the catch.